fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordContext.java

```diff
@@ -435,6 +438,7 @@ public void commit() {
     public CompletableFuture<Void> commitAsync() {
         final long startTimeNanos = System.nanoTime();
         ensureActive();
+        mergeLuceneIndexIfNeeded();
```

Comment:
Rather than having the logic for only issuing one merge live here in `FDBRecordContext`, it seems like it would be better if the logic lived in the commit hook instead. Given that the `FDBRecordContext` already supports named commit checks, could you do something like:

1. Have a named commit check called something like "LUCENE_EXECUTE_MERGE" or something like that
2. It has an implementation that has a list of directories that need merging
3. In the merge implementations, the merge commit check is read, and then the merge operation is added to the queue
4. At commit check execution time, the commit check itself handles picking and executing one of the merges

I think this would allow all of the Lucene-specific logic to leave the `FDBRecordContext`